app/assets/javascripts/app/core/keyboard.js

```diff
@@ -0,0 +1,30 @@
+(function() {
+  'use strict';
+
+  angular
+    .module('council.core')
+    .service('Keyboard', Keyboard);
+
+  function Keyboard($document) {
+    var input = $('<input type="text">');
+
+    function gainFocus() {
+      $($document[0].body).append(input);
+
+      input.focus();
+    }
+
+    function loseFocus() {
+      $(input).remove();
+    }
+
+    function close() {
+      _.debounce(gainFocus, 50);
+      _.debounce(loseFocus, 50);
```

Comment:
@gabrielpoca I finally know why I need this. Since we have two directives with the same controller we'll have two instances of the controller, causing the event to be listened two times and each of them will call the function. The easiest way to prevent this is to debounce the calls.

What do you think?